### PR TITLE
[flutter_tools] restore report timings functionality to build aot

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

This functionality is still used by golem. Thankfully all of the relevant information is still present, with a little massaging for format.

### CODE REVIEW

File: packages/flutter_tools/lib/src/aot.dart [hunk 1]
```diff
@@ -106,6 +107,20 @@ class AotBuilder {
...
+      final PerformanceMeasurement kernel = result.performance['kernel_snapshot'];
+      PerformanceMeasurement aot;
+      if (expectSo) {
+        aot = result.performance.values.firstWhere(
+          (PerformanceMeasurement measurement) => measurement.analyicsName == 'android_aot');
+      } else {
+        aot = result.performance.values.firstWhere(
+          (PerformanceMeasurement measurement) => measurement.analyicsName == 'ios_aot');
+      }
+      globals.printStatus('frontend(CompileTime): ${kernel.elapsedMilliseconds} ms.');
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/aot.dart]: Consider adding a test that this output appears in the logs. Consider adding a note here to consult with the Dart team before removing or modifying.

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/aot.dart]: Done

File: packages/flutter_tools/test/general.shard/commands/build_aot_test.dart [hunk 2]
```diff
@@ -106,6 +108,48 @@ void main() {
     PlistParser: () => mockPlistUtils,
   });
 
+   testUsingContext('build aot outputs timing info', () async {
+     globals.fs.file('.dart_tool/flutter_build/cce09742720db17ffec62331bd7e42d5/app.so')
+      .createSync(recursive: true);
```
Comment (zanderso) [inline packages/flutter_tools/test/general.shard/commands/build_aot_test.dart]: nit: 2 space indent.

File: packages/flutter_tools/test/general.shard/commands/build_aot_test.dart [hunk 3]
```diff
@@ -106,6 +108,48 @@ void main() {
...
+     when(buildSystem.build(any, any))
+      .thenAnswer((Invocation invocation) async {
```
Comment (zanderso) [inline packages/flutter_tools/test/general.shard/commands/build_aot_test.dart]: ditto

File: packages/flutter_tools/test/general.shard/commands/build_aot_test.dart [hunk 4]
```diff
@@ -106,6 +108,48 @@ void main() {
...
+        });
+      });
+
+     await AotBuilder().build(
+       platform: TargetPlatform.android_arm64,
+       outputPath: '/',
+       buildInfo: BuildInfo.release,
+       mainDartFile: globals.fs.path.join('lib', 'main.dart'),
+       reportTimings: true,
+      );
```
Comment (zanderso) [inline packages/flutter_tools/test/general.shard/commands/build_aot_test.dart]: ditto.

**Review by zanderso — APPROVED** [review]
lgtm w/ nits

File: packages/flutter_tools/test/general.shard/commands/build_aot_test.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/commands/build_aot_test.dart]: Woops, not sure how that happened

**Pull request merged**